src/main/java/com/google/cloud/genomics/dataflow/readers/bam/Reader.java

```diff
@@ -44,22 +44,38 @@
   BAMShard shard;
   DoFn<BAMShard, Read>.ProcessContext c;
   Stopwatch timer;
-  ValidationStringency stringency;
+  ReaderOptions options;
   
   SAMRecordIterator iterator;
-  boolean procesingUnmapped;
+  enum Filter {
+    UNMAPPED_ONLY,
+    MAPPED_AND_UNMAPPED,
+    MAPPED_ONLY
+  }
+  
+  Filter filter;
   
   int recordsBeforeStart = 0;
   int recordsAfterEnd = 0;
   int mismatchedSequence = 0;
   int recordsProcessed = 0;
   
-  public Reader(Objects storageClient, ValidationStringency stringency, BAMShard shard, DoFn<BAMShard, Read>.ProcessContext c) {
+  public Reader(Objects storageClient, ReaderOptions options, BAMShard shard, DoFn<BAMShard, Read>.ProcessContext c) {
     super();
     this.storageClient = storageClient;
     this.shard = shard;
     this.c = c;
-    this.stringency = stringency;
+    this.options = options;
+    filter = setupFilter(options, shard.contig.referenceName);
+  }
+  
+  public static Filter setupFilter(ReaderOptions options, String referenceName) {
+    if (referenceName.equals("*")) {
```

Comment:
I was a little surprised to see this line. Can you add some context (in a comment) on when this would happen? It looks like it would come from the header? When would a header have * contig?